### ISSUE TITLE
New update broke some proton compatibility with games.

### ISSUE THREAD

**Issue by freddyyeddy** [ChimeraOS/chimeraos#247]
All of the lego games were working fine with using proton before the update now they crash. Im trying every version now to see if they will start working under a new version. But at present the load shaders then i wait for it to start and it crashes unceremoniously. I dont know if there is any log file i could use to diagnose whats happening but if there is I would be happy to share. Im sure the lego games are not the only proton compatibility games that got borked.

**Comment by alkazar** [comment]
Could you provide a list of the games that you found no longer work?

**Comment by freddyyeddy** [comment]
Lego DC vilians, and Lego Jurassic park. Those are the only ones I've
noticed. I've tried every version of proton that is preloaded not a one
will load. Just white screen crash back to steam.

**Comment by alkazar** [comment]
I was able to try LEGO DC Super-Villains and it works great with Proton 6.3. I do not have LEGO Jurassic World, but was also able to test LEGO Marvel Super Heroes.

What GPU are you using?
Do any other games work?

**Comment by freddyyeddy** [comment]
I'm not at home presently but Im using an alienware alpha. Can't remember
specs off the top of my head I'll follow up when I get home in a week with
system specs.

**Comment by freddyyeddy** [comment]
Ok just got home.
 GPU is Nvidia GeForce SE22?  2 gigs of ram Using driver 4.6.0 NVIDIA 465.3 (all from steam page)
 According to the marketing the Alienware alpha has "Custom Nvidia “Maxwell” GTX GPU w/2GB GDDR5 running @6Gbps"
 CPU 2 Core, 4 Threads With 8 Gigs Ram
 According to the marketing the Alienware Alpha has "4th Generation Intel Core i "Haswell" Dual and Quad Core CPU support"
 
 As far as proton 6.3 working no dice. other games are working fine. Katamari Dimacy (another proton game) works fine, also other linux native games have been working ok. Could be related but retorarc sometimes works sometimes doesn't depending on updates too (roms that worked fine before majoras mask, Pokémon snap for example). My guess is could be graphics card related. Tell me what you need on my end and I will be happy to oblige. I appreciate your time and efforts

**Comment by Samsagax** [comment]
Maybe proton is picking up your Intel graphics card instead of Nvidia one? Have you tried disabling the ICD loaders for Intel? Is known that Vulkan support for Haswell and prior is not complete.

Try this:
- look into `/usr/share/vulkan/icd.d/` for Intel related `.json` files.
- rename them to something ICD loaders won't load like `intel_icd.x86_64.json.backup`
- make sure Nvidia loader is there.
- try again launching the game

If the avobe worked, you may want to select your Vulkan driver globally by setting 
`VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.i686.json:/usr/share/vulkan/icd.d/nvidia_icd.x86_64.json`
in your `.profile`. Your mileage may vary but take a look into [the ArchWiki](https://wiki.archlinux.org/title/Vulkan)

**Comment by freddyyeddy** [comment]
Ok so ive tried everything I could but keep getting read only file system error "mv: cannot move 'intel_icd.i686.json' to 'intel_icd.i686.json.back': Read-only file system". This could be the issue I've re-booted several times and cannot get it to go into rw filesystem here are the tricks ive tried so far to no avail. 
 "sudo mount -n -o remount,rw /"
 "sudo fsck -Af -M"
 here is the journalctl -k output in a gist [Link Here](https://gist.github.com/freddyyeddy/9de28b742548014f2a730f2603b8b8ad)

**Comment by Samsagax** [comment]
> Ok so ive tried everything I could but keep getting read only file system error "mv: cannot move 'intel_icd.i686.json' to 'intel_icd.i686.json.back': Read-only file system". This could be the issue I've re-booted several times and cannot get it to go into rw filesystem here are the tricks ive tried so far to no avail.
> "sudo mount -n -o remount,rw /"
> "sudo fsck -Af -M"
> here is the journalctl -k output in a gist [Link Here](https://gist.github.com/freddyyeddy/9de28b742548014f2a730f2603b8b8ad)

That's because the root filesystem is mounted read only. You should try setting

`VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.i686.json:/usr/share/vulkan/icd.d/nvidia_icd.x86_64.json`

Before game command (make sure the paths exists, I'm writing this from memory)

**Comment by freddyyeddy** [comment]
So just to make sure I understand you mean set that in the launch options
for each game with the issue? or just run that in the SSH terminal? if its
the ladder i tried that too but they still aren't working even swapped to
the version of proton you suggested for the Lego villains game still loads
shaders then white screen and crash back to steam.

**Comment by Samsagax** [comment]
Set it in the launch options of a game you would like to test before the `%command%` bit. If it works you can add it permanently to your session globally for all games by adding it to your `.profile` or in `~/.config/steamos-compositor-plus` files or by adding it via systemd (recommended): https://wiki.archlinux.org/title/systemd/User#Environment_variables

**Comment by freddyyeddy** [comment]
Ok thats clear as mud. The Arch rabbit hole for setting systemd is
confounding to me. but i can figure that out for the sake of making these
games function. But there is no %command% in launch options, just a blank
box. So don't understand what i should type
there VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.i686.json:/usr/share/vulkan/icd.d/nvidia_icd.x86_64.json?
Trying to understand what the heck i need to do to even test this. Ive got
enough linux experience to figure out ssh and stuff. but linux filesystem
stuff is beyond my skillset. Im willing to figure it out just need some
more guidance. I really appreciate your time and efforts.

**Comment by Samsagax** [comment]
I'll try to be more clear and don't assume knowledge.

First of all you are on track, you found the launch options. To know how they work, here is the wiki entry; https://wiki.archlinux.org/title/Steam#Launch_options

So in your case you should fill in: `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.i686.json:/usr/share/vulkan/icd.d/nvidia_icd.x86_64.json %command%`


Test a proton game and report back. If that works, then we can move forward to the next step.

**Comment by freddyyeddy** [comment]
Got a ""Sorry, but your machine is unable to run the game. It requires a
graphics card which supports directx 11." Whitch is weird because i checked
before and those file locations were correct. Went to check now
and..... [image:
image.png]  shows radion instead of nvidia so dont know how i effed this up
but going to await further instruction. I appreciate it. I had
added VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.i686.json:/usr/share/vulkan/icd.d/nvidia_icd.x86_64.json
to the steam compositor before but that shoudlnt have borked it right?
removed the line and rebooted still radion instead of nvidia.

**Comment by Samsagax** [comment]
That's even weirder...

How radeon appeared there?

**Comment by freddyyeddy** [comment]
I dont even know last time i checked literally an hour or so prior it only
had the nvidia ones. can i overwrite these... wait i cant its read only
even with pseudo. Is there any way ssh wise to force a reinstall of that
part of OS without overwriting steam games ect. or should I just do a fresh
install from scratch (i don't want to but am willing to)? the weird thing
with the read only part is I can save games ect all progress on my games
has been saved just fine. My uneducated guess is there is/was some issue
with the recent update that messed up some system file stuff and makes the
system start in read only (something arch is known to do in rare occasions)
and borks the proton stuff due to that. This is all above my depth and
basically a wild barley educated guess. Let me know what I should do on my
end and I can run whatever diagnostics or patches you suggest so try and
ferret this out. I'm going back to my remote work for another week and
won't be able to do anything with this system until next friday night at
earliest. Again I appreciate your time and expertise.

**Comment by Samsagax** [comment]
Let's try this:

Add
````
export __NV_PRIME_RENDER_OFFLOAD=1
export __VK_LAYER_NV_optimus=NVIDIA_only
export __GLX_VENDOR_LIBRARY_NAME=nvidia
````

To `~/.config/steamos-compositor-plus` (you may need to create it)

Then remove all other workarounds and tweaks y may have done.
Reboot your PC. And try again to launch the game.

Regarding reinstalling you can force a reinstall with `$ sudo frzr-deploy chimeraos/chimeraos:stable` in a terminal. But that won't fix your issue. The root filesystem is read-only by default except for `/home` and `/etc` so you shouldn't be able to modify anything outside that scope. Games, configuration and saves are all inside your home directory.

**Comment by freddyyeddy** [comment]
Black screen with flecks on load. randomly hit buttons and loaded a game
and heard it load but no visible screen.

**Comment by linuxgaming** [comment]
I'm having a similar problem and tried some of these fixes to no avail. One thing I noticed is that you recommend to use "VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.i686.json:/usr/share/vulkan/icd.d/nvidia_icd.x86_64.json %command%" but I don't have nvidia_icd.i686.json and nvidia_icd.x86_64.json - just nvidia_icd.json - while intel and radeon icd files do have x86_64 and i686 versions, is that normal?

**Comment by Samsagax** [comment]
> I'm having a similar problem and tried some of these fixes to no avail. One thing I noticed is that you recommend to use "VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.i686.json:/usr/share/vulkan/icd.d/nvidia_icd.x86_64.json %command%" but I don't have nvidia_icd.i686.json and nvidia_icd.x86_64.json - just nvidia_icd.json - while intel and radeon icd files do have x86_64 and i686 versions, is that normal?

This kind of problem can have multiple sources. Proton relies on vulkan support from the graphics card. What GPU do you have? Does starting with `PROTON_LOG=1 %command%` produces a log? Which games are you trying?

**Comment by linuxgaming** [comment]
I have nvidia GTX 970. We tried many games, for example Boomerang Fu (used to work perfectly until a few days ago). The logs work but don't produce anything immediately useful. Attached is the photo of the top part, I have no easy way to transfer the actual file from that computer.
![photo_2021-08-05_22-04-05](https://user-images.githubusercontent.com/42684928/128413872-ed278e56-9155-415e-8af3-806966d94dc9.jpg)

**Comment by Samsagax** [comment]
You can transfer files using `ssh` and `scp` (after you configure your keys in the web app)
I don't see where the issue arises in that partial log. But I can see you are running from `/mnt/` Proton needs write access to the drive to work properly. Maybe using a game from the default library works? Could you test that?
Also what are the permissions on that directory (`/mnt/steamapps`)?

**Comment by linuxgaming** [comment]
It's the default library, it's just symlinked. I've been using it without issues since 2013.

**Comment by Samsagax** [comment]
> It is the default library, it's just symlinked (it's on another disk). I've been using it without issues since 2013. All the games are there, not just proton games.

You didn´t answer my question. What are the permissions on that directory? What happens with a game installed under `~/.local/share/Steam/steamapps/common` with proton?
